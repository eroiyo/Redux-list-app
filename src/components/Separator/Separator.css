.inside {
  background-color: #35548b;
  display: flex;
  width: 100%;
  height: 52%;
  padding-top: 2%;
  padding-bottom: 2%;
}

.Separator {
  width: 100%;
  height: 5%;
  background-color: #35548b;
}
