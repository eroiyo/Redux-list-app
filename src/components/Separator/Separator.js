import './Separator.css';
import React from 'react';
import Proptypes from 'prop-types';

function Separator(props) {
  const { show } = props;
  if (show === undefined) {
    return (
      <div className="Separator">
        <div className="inside">
          <h6 className="header-title separator-title">Categorys</h6>
        </div>
      </div>
    );
  }
  return (
    <div className="Separator">
      <div className="inside">
        <h6 className="header-title separator-title">{show.cat}</h6>
      </div>
    </div>
  );
}

Separator.displayName = 'Displayer';

Separator.defaultProps = {
  show: undefined,
};

Separator.propTypes = {
  show: Proptypes.shape(
    {
      cat: Proptypes.string,
      total: Proptypes.number,
      img: Proptypes.string,
    },
  ),
};
export default Separator;
