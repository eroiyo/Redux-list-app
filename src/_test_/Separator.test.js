import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Proptypes from 'prop-types';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Separator from '../components/Separator/Separator';

const MockSeparator = ({ show }) => (
  <BrowserRouter>
    <Provider store={store}>
      <Separator show={{ show }} />
    </Provider>
  </BrowserRouter>
);

MockSeparator.propTypes = {
  show: Proptypes.shape(
    {
      cat: Proptypes.string,
      total: Proptypes.number,
      img: Proptypes.string,
    },
  ),
};

MockSeparator.defaultProps = {
  show: undefined,
};

describe('Test the Separator component', () => {
  beforeEach(() => {
    render(<MockSeparator show={{ cat: 'Jewelery' }} />);
  });

  it('should match the snapshot', () => {
    const tree = renderer.create(<MockSeparator year="2021" title="Most views" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
