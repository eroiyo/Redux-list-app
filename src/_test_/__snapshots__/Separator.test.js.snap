// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Separator component should match the snapshot 1`] = `
<div
  className="Separator"
>
  <div
    className="inside"
  >
    <h6
      className="header-title separator-title"
    />
  </div>
</div>
`;
